src/reactjs/useIsMutating.ts

```diff
@@ -2,22 +2,32 @@ import React from 'react'
 
 import { notifyManager } from '../core/notifyManager'
 import { MutationKey } from '../core/types'
+import { QueryClient } from '../core/queryClient'
 import { MutationFilters, parseMutationFilterArgs } from '../core/utils'
 import { useQueryClient } from './QueryClientProvider'
 
-export function useIsMutating(filters?: MutationFilters): number
+interface Options {
+  context?: React.Context<QueryClient | undefined>
+}
+
+export function useIsMutating(
+  filters?: MutationFilters,
+  options?: Options
+): number
 export function useIsMutating(
   mutationKey?: MutationKey,
-  filters?: Omit<MutationFilters, 'mutationKey'>
+  filters?: Omit<MutationFilters, 'mutationKey'>,
+  options?: Options
 ): number
 export function useIsMutating(
   arg1?: MutationKey | MutationFilters,
-  arg2?: Omit<MutationFilters, 'mutationKey'>
+  arg2?: Omit<MutationFilters, 'mutationKey'> | Options,
+  arg3?: Options
 ): number {
   const mountedRef = React.useRef(false)
-  const filters = parseMutationFilterArgs(arg1, arg2)
+  const [filters, options = {}] = parseMutationFilterArgs(arg1, arg2, arg3)
 
-  const queryClient = useQueryClient()
+  const queryClient = useQueryClient({ context: (options as Options).context })
```

Comment:
why is the type assertion here necessary, while it is not necessary for `useIsFetching` ?

Reply:
Great point. Pushed a new commit that aligns `parseFilterArgs` with `parseMutationFilterArgs`. For some reason, their types were not aligned even though their functionality is identical.